src/hooks/useActionCenterSubscriber/useEvergreenDepositActionSubscriber.tsx

```diff
@@ -0,0 +1,85 @@
+import { usePrevious, useToast } from '@chakra-ui/react'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useEffect } from 'react'
+
+import { useWallet } from '../useWallet/useWallet'
+
+import { EvergreenDepositNotification } from '@/components/Layout/Header/ActionCenter/components/Notifications/EvergreenDepositNotification'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import { selectPendingEvergreenDepositActions } from '@/state/slices/actionSlice/selectors'
+import { ActionStatus } from '@/state/slices/actionSlice/types'
+import { selectTxs } from '@/state/slices/selectors'
+import { serializeTxIndex } from '@/state/slices/txHistorySlice/utils'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+type UseEvergreenDepositActionSubscriberProps = {
+  onDrawerOpen: () => void
+  isDrawerOpen: boolean
+}
+
+export const useEvergreenDepositActionSubscriber = ({
+  onDrawerOpen,
+  isDrawerOpen,
+}: UseEvergreenDepositActionSubscriberProps) => {
+  const dispatch = useAppDispatch()
+  const {
+    state: { isConnected },
+  } = useWallet()
+
+  const toast = useToast({
+    duration: isDrawerOpen ? 5000 : null,
+    position: 'bottom-right',
+  })
+
+  const previousIsDrawerOpen = usePrevious(isDrawerOpen)
+
+  useEffect(() => {
+    if (isDrawerOpen && !previousIsDrawerOpen) {
+      toast.closeAll()
+    }
+  }, [isDrawerOpen, toast, previousIsDrawerOpen])
+
+  const pendingEvergreenActions = useAppSelector(selectPendingEvergreenDepositActions)
```

Comment:
That's absolutely fine for the time being, but this won't scale, we should probably think one more level of abstraction here and have a hook that does generalised Txs introspection (it itself *could* leverage multiple lookup hooks)

Reply:
Also re: the question of inserting from where staking happens, and updating here, not sure about this one either.

It doesn't hurt to, but feels like it adds overhead, we could simply select Txs filtered by `to` being RFOX contracts (see `maybeRefetchOpportunities` in `src/hooks/useTransactionsSubscriber.ts` for an example of where we do that)

Reply:
Closing spewy wall of text